src/lightning_app/core/app.py

```diff
@@ -184,7 +184,8 @@ def __init__(
     def _update_index_file(self):
         # update index.html,
         # this should happen once for all apps before the ui server starts running.
-        frontend.update_index_file(FRONTEND_DIR, info=self.info, root_path=self.root_path)
+        if self.root_path != "":
```

Comment:
This is already handled in the `update_index_file` - but that also does some other things so this could cause errors

Reply:
Yes, better to be safe.